runtime/include/http_session.h

```diff
@@ -419,11 +371,21 @@ http_session_receive_request(struct http_session *session, void_star_cb on_eagai
 			goto err;
 
 		assert(bytes_received > 0);
-		assert(session->request_buffer.length < session->request_buffer.capacity);
 
-		session->request_buffer.length += bytes_received;
+		/* Write temp buffer to memstream */
+		fwrite(temp, 1, bytes_received, session->request_buffer.handle);
+
+		/* fflush memstream managed buffer */
+		fflush(session->request_buffer.handle);
 
-		ssize_t bytes_parsed = http_session_parse(session, bytes_received);
+		/* Invalidate parser structure if buffer moved */
+		ssize_t bytes_parsed;
+		if (old_buffer != session->request_buffer.data) {
+			http_session_parser_init(session);
+			bytes_parsed = http_session_parse(session, session->request_buffer.size);
+		} else {
+			bytes_parsed = http_session_parse(session, bytes_received);
```

Comment:
Looking at the implementation of http_session_parse the second argument seems to be useless. So, this if clause go away ?

Reply:
Ended up actually using it for the updated version of the http_session_parse function (line:304)